src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -33,13 +34,41 @@ public class NodeReadConverter : JsonConverter
     {
         private CustomNodeManager manager;
         private LibraryServices libraryServices;
+        private bool isTestMode;
+
         
         public ElementResolver ElementResolver { get; set; }
+        //map of all loaded assemblies including LoadFrom context assemblies
+        private Dictionary<string, List<Assembly>> loadedAssemblies;
 
-        public NodeReadConverter(CustomNodeManager manager, LibraryServices libraryServices)
+        public NodeReadConverter(CustomNodeManager manager, LibraryServices libraryServices, bool isTestMode = false)
         {
             this.manager = manager;
             this.libraryServices = libraryServices;
+            this.isTestMode = isTestMode;
+
+            if (this.isTestMode)
+            {
+                this.loadedAssemblies = this.buildMapOfLoadedAssemblies();
+            }
+        }
+
+        private Dictionary<string,List<Assembly>> buildMapOfLoadedAssemblies()
+        {
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var dict = new Dictionary<string, List<Assembly>>();
+            foreach(var assembly in allAssemblies)
+            {
+                var resultList = new List<Assembly>();
+                if (!dict.TryGetValue(assembly.GetName().Name, out resultList))
+                {
```

Comment:
resultList is not used here. 
Also, is it doing something like this ? ```allAssemblies.select(x=>x.GetName().Name).Distinct()```
Then  ```dict[assembly.GetName().Name].Add(assembly)```.

Reply:
we're building a list of all assemblies which share the same name - there are some assemblies which we load multiple versions of with the same name - it's not really an issue as those are not dlls we are testing but we need to avoid the exception that is thrown when adding multiple keys.

I'll try to get rid of resultList - I can't remember if it's required or optional for tryGetValue.

Reply:
using distinct on the list should do that?

Reply:
I'm not clear how I would then get a list of assemblies associated with each name. what are you thinking?